### DISCUSSION TITLE
Extract_fa does not extract half adder

### DISCUSSION THREAD

**Discussion by viyelesay** [YosysHQ/yosys#2956] [Q&A]
extract_fa command extracts full adders but does not extract half adder. Yosys uses full adder instead of half addder and connects carry in input to zero. Please check the attached file for the issue.
[adderissue.zip](https://github.com/YosysHQ/yosys/files/5155170/adderissue.zip)

**Comment by whitequark** [comment]
Yosys does not have a half adder cell, so the pass uses a `$fa` cell to represent half adders.

**Comment by viyelesay** [comment]
Thanks for your reply. It would be nice if it also supports $fa cell.
